src/main/java/io/github/eocqrs/qce/DomainObject.java

```diff
@@ -0,0 +1,16 @@
+package io.github.eocqrs.qce;
+
+/**
+ * The simple domain object
+ *
+ * @param <E> The event type to produce
+ */
+public interface DomainObject<E extends Event> {
+
+    /**
+     * Domain object as event
+     *
+     * @return Domain object transformed to event
+     */
+    E asEvent();
```

Comment:
@l3r8yJ what is the purpose of `asEvent()`?

```java
User user = ...;
```
and then we have UserRegisteredEvent, UserUpdatedEvent and so on
which kind of event User will represent, if it extends `DomainObject`?

Can you explain what is the purpose of the `DomainObject` interface, if we already have `Aggregate` interface, that represents chain of Events

Reply:
@h1alexbel i think you give me some idea